### ISSUE TITLE
scylla doesn't survive the "soak test" and is being killed

### ISSUE THREAD

**Issue by vladzcloudius** [scylladb/scylla#171]
Running an endurance test that opens a lot of cassandra-stress clients in parallel that run for 10h each (a.k.a. "soak test") ends up with OS killing a scylla process (after less than an hour). 
The test is well described in the issue #142.

The master HEAD used: c6ea25c5fbfc88e7e04ca8b35f55bf0f9d014213

While running the test one may clearly see how the amount of memory used by scylla increases all the time.

dmesg shows the following:
```
[970235.086838] Out of memory: Kill process 10101 (scylla) score 715 or sacrifice child                                   41 root      20   0       0      0      0 S   0.0  0.0   2:15.35 rcuos/4
[970235.086872] Killed process 10101 (scylla) total-vm:17188401380kB, anon-rss:25427916kB, file-rss:32040kB 
```

Running in a debug mode didn't show a lot except for standard "use-after-free" stuff which doesn't seem to be related.

The issue reproduces both with DPDK and Posix network backends.

The issue didn't reproduce with the HEAD  0b3d2de

**Comment by tzach** [comment]
@avikivity @vladzcloudius is this still open?

**Comment by avikivity** [comment]
Should be fixed now (01c3670defcb4144 and others).
